### PR TITLE
refactor: Removes the deprecated VERSIONED_EXPORT feature flag

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
As part of the [4.0 approved initiatives](https://github.com/orgs/apache/projects/292/views/1), this PR removes the `VERSIONED_EXPORT` feature flag.

The previous value of the feature flag was `True` and now the feature is permanently enabled.

### TESTING INSTRUCTIONS
CI should be sufficient for merging this PR. We'll do a complete testing of 4.0 after all approved proposals are merged.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by eschutho** [comment]
@michael-s-molina should we put in the PR description and UPDATING.md what the flag currently defaults to? I think that would help people know what behavior to expect.

File: superset/cli/importexport.py [hunk 1]
```diff
@@ -60,338 +58,133 @@ def import_directory(directory: str, overwrite: bool, force: bool) -> None:
...
-    @click.command()
-    @with_appcontext
-    @click.option(
-        "--back-references",
-        "-b",
-        is_flag=True,
-        default=False,
-        help="Include parent back references",
-    )
-    def export_datasource_schema(back_references: bool) -> None:
```
Comment (eschutho) [inline superset/cli/importexport.py:391]: @michael-s-molina was this removed?

File: superset/cli/importexport.py [hunk 2]
```diff
@@ -60,338 +58,133 @@ def import_directory(directory: str, overwrite: bool, force: bool) -> None:
...
-        "-p",
-        help="Path to a single JSON file or path containing multiple JSON "
-        "files to import (*.json)",
-    )
-    @click.option(
-        "--recursive",
-        "-r",
-        is_flag=True,
-        default=False,
-        help="recursively search the path for json files",
```
Comment (eschutho) [inline superset/cli/importexport.py:294]: Were some options removed? I think we may still need the import from json to allow people to migrate from an older version still?

**Review by justinpark — APPROVED** [review]
LGTM

**Pull request merged**